### PR TITLE
Improvements to finding corner points in `IdCoordinateSource`.

### LINKED ISSUE #1016: Improvements to finding corner points in `IdCoordinateSource`
While working on [this](https://github.com/ie3-institute/simona/pull/724) SIMONA PR, I came up with this idea.

### PR DESCRIPTION
**Pull request by staudtMarius**
Resolves #1016

### CODE REVIEW

**Review by sebastian-peter — COMMENTED** [review]
Hmm. To be honest, I don't get the point. You're separating the functionality into two methods - ok, fine. But this is not more efficient, actually. 

Now we're calculating `distances.stream().filter(c -> !corners.contains(c))` on top of what we've been doing before. This had been part of the `List<CoordinateDistance> other` before.

**Comment by staudtMarius** [comment]
My idea with this PR is to have the part of finding the corners to be accessible from outside.
But I see your point. I changed it with the next commit.

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Just one request from my side for now, will look deeper into it later at some point

**Review by danielfeismann — COMMENTED** [review]
LGTM, just one question to the docs...

File: docs/readthedocs/models/input/additionaldata/idcoordinatesource.md [hunk 1]
```diff
@@ -7,11 +7,21 @@ coordinates.
...
 ## Information
 
-| Attribute    | Remarks                                                        |
-|:-------------|:---------------------------------------------------------------|
-| `Id`         | An integer value for identifying the coordinate.               |
-| `Coordiante` | Geographical information presented as `Lat/long` of a `Point`. |
+```{eval-rst}
+.. list-table::
+   :widths: 33 33
+   :header-rows: 0
```
[the remark this answers is not in the document]
Comment (staudtMarius) [reply docs/readthedocs/models/input/additionaldata/idcoordinatesource.md:13]: It is zero in all the other docs too. Also I just tested changing this and I couldn't see any difference. So I think this should be fine.

[the remark this answers is not in the document]
Comment (danielfeismann) [reply docs/readthedocs/models/input/additionaldata/idcoordinatesource.md:13]: Trust me ;) the tables look nicer then. Would be great if you can fix this at the other tables as well.

[the remark this answers is not in the document]
Comment (staudtMarius) [reply docs/readthedocs/models/input/additionaldata/idcoordinatesource.md:13]: Ok. I will adapt these parts.

**Review by danielfeismann — APPROVED** [review]
Looks all good to me, thanks for your work done here!

**Pull request merged**